### PR TITLE
Adds lame duck mode

### PR DESCRIPTION
**Pull request by autodidaddict**

### CODE REVIEW

File: internal/node/node.go [hunk 1]
```diff
@@ -135,6 +136,23 @@ func (n *Node) Stop() {
 	n.shutdown()
 }
 
+func (n *Node) EnterLameDuck() error {
+	atomic.StoreUint32(&n.lameduck, 1)
```
Comment (kthomas) [inline internal/node/node.go:140]: Should there be a check here to make sure `n.lameduck` isn't already 1 to prevent accidentally calling it twice?

Comment (kthomas) [reply internal/node/node.go:140]: Currently it seems `procMan.EnterLameDuck()` is idempotent for both of the implementations but it seems like a bad assumption that they will always behave that way.

Comment (autodidaddict) [reply internal/node/node.go]: good call. made it only happen once.

**Review by kthomas — DISMISSED** [review]

**Comment by kthomas** [comment]
This just needs to be rebased on main to fix the linting issue.

**Review by jordan-rash — APPROVED** [review]

**Pull request merged**